examples/sample.env

```diff
@@ -0,0 +1 @@
+REACT_APP_API_URL='https://dev-api.clearlydefined.io'
```

Comment:
We should have preconfigured .env files for local, dev and prod. Forcing a user know about and create these files is error prone and adds friction. For this scenario (website) we do not have any secrets so its quite reasonable to put the files in the repo.

side note, in effect we have just moved the code that was in `clearlydefined.js` out into the config and build steps. The new approach is more extensible etc so good.

Reply:
@jeffmcaffer I ran a few days ago with @storrisi  through this. While right now we don't have any Secrets, we should still keep the configuration outside the repo. As the application grows we might end up needing to add new configurations (some of which env specific) and starting with the right foot seems like a good idea (also you get security warnings as you commit .env files, and IMHO we should lead by example and stick to security standards). 

As you can see, we still have added samples + documentation to help developers in understanding the correct usage.

Reply:
I'm certainly sympathetic to security concerns but I'll argue that as a design principle we explicitly want to NOT have secrets in the web site. It is just a generated static site so secrets would then be embedded in static downloadable content. If we do have secrets then we will have had some major architecture shifts and how to populate secrets will be part of that. We should optimize for simplicity now and add complexity as/if needed. 

Having to add more documentation to help users understand is another red flag.

If having .env files checked in causes security warnings (like what and from what system), then we should do something different or turn them off.